middleware/index.ts

```diff
@@ -282,6 +284,58 @@ function errorMessage(error: Error): any {
         .build();
 }
 
+function isInitialAsyncRequest(payload: any, headers: any): boolean {
+    return payload.type 
+        && payload.type.startsWith('com.salesforce.function.async')
+        && !headers[ASYNC_HEADER];
+}
+
+// Invoke given function again to fulfill async request; response is not handled
+async function invokeAsyncFn(logger: Logger, payload: any, headers: any): Promise<void> {
+    let hostUrl = headers['Host'] || payload?.data?.sfContext?.resource;
+    if (!hostUrl) {
+        throw new Error('Unable to determine host');
+    }
+
+    headers[ASYNC_HEADER] = 'true';
+    const parsedUrl = require('url').parse(hostUrl);
```

Comment:
I'd suggest moving `require(`url`)` at the top of the file. A single instance of the middleware is held onto for the duration of the pod's life. `require` can be expensive sometimes.

Reply:
`require` caches, right?  No problem to move to top.  Can do same for `http/s` lib.

Reply:
Yeah, it does cache. I guess it really matters where we want to pay for the initial require cost.